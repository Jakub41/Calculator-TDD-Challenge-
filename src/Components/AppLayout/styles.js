import styled from "@emotion/styled";

export const ContentLayout = styled.div`
	width: 100px;
	height: 100px;
`
