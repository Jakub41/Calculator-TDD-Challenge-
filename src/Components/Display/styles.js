import styled from "@emotion/styled";

export const DisplayDiv = styled.div`

`;
