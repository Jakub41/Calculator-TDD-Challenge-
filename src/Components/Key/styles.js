import styled from "@emotion/styled";

export const KeyDiv = styled.div`

`;

export const KeyButton = styled.button`

`;

