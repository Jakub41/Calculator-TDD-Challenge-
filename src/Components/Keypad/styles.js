import styled from "@emotion/styled";

export const KeypadDiv = styled.div`

`;

export const NumbersDiv = styled.div`

`;

export const OperatorsDiv = styled.div`

`;
