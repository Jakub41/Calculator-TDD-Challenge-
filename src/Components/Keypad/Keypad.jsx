import React from "react";
import PropTypes from "prop-types";
import { KeypadDiv, NumbersDiv, OperatorsDiv } from "./styles";

const Keypad = ({
	callOperator,
	numbers,
	operators,
	setOperator,
	updateDisplay,
}) => (
	<KeypadDiv>
		<NumbersDiv></NumbersDiv>
		<OperatorsDiv></OperatorsDiv>
	</KeypadDiv>
);

// Keypad.propTypes = {
// 	callOperator: PropTypes.func.isRequired,
// 	numbers: PropTypes.array.isRequired,
// 	operators: PropTypes.array.isRequired,
// 	setOperator: PropTypes.func.isRequired,
// 	updateDisplay: PropTypes.func.isRequired,
// };

export default Keypad;
