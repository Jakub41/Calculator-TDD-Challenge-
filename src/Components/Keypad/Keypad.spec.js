import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Keypad from "./Keypad";
import { KeypadDiv, NumbersDiv, OperatorsDiv } from "./styles";

describe("Keypad", () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(
			<Keypad
			// callOperator={jest.fn()}
			// numbers={[]}
			// operators={[]}
			// setOperator={jest.fn()}
			// updateDisplay={jest.fn()}
			/>
		);
	});

	it("should render correctly", () => expect(wrapper).toMatchSnapshot());

	it("should render Keypad block", () => {
		const tree = renderer
			.create(
				<KeypadDiv>
					<NumbersDiv></NumbersDiv>
					<OperatorsDiv></OperatorsDiv>
				</KeypadDiv>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("should render a <KeypadDiv />", () => {
		expect(wrapper.find(KeypadDiv)).toHaveLength(1);
	});

	it("should render a <NumbersDiv />", () => {
		expect(wrapper.find(NumbersDiv)).toHaveLength(1);
	});

	it("should render a <OperatorsDiv />", () => {
		expect(wrapper.find(OperatorsDiv)).toHaveLength(1);
	});
});
