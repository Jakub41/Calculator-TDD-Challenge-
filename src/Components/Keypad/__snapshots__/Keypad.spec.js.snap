// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad should render Keypad block 1`] = `
<div
  className="css-1esthja"
>
  <div
    className="css-1esthja"
  />
  <div
    className="css-1esthja"
  />
</div>
`;

exports[`Keypad should render correctly 1`] = `
<Styled(div)>
  <Styled(div) />
  <Styled(div) />
</Styled(div)>
`;
